openwisp_controller/config/static/config/js/device.js

```diff
@@ -0,0 +1,37 @@
+"use strict";
+django.jQuery(function ($) {
+  if ($(".add-form").length || !$("#device_form").length) {
+    return;
+  }
+
+  // replaces the management ip field with text with the option to edit it
+  var ip_input = $(".field-management_ip > div > input");
+  var initial_ip = ip_input.val();
+  ip_input.after(function () {
+    ip_input.hide();
+    return (
+      '<span class="readonly" id="management_ip_text">' +
+      (initial_ip === "" ? "-" : initial_ip) +
+      "</span>"
+    );
+  });
+  $("#management_ip_text").after(
+    '<a id="edit_management_ip"><img src="/static/admin/img/icon-changelink.svg" alt="Edit"></a>'
```

Comment:
please use `gettext('Edit')` which flags the string as translatable.
Please also add a `title` attribute with the same string (saving the gettext() result in a var which is used in both attributes) so that a tooltip is shown to the user on hover.